### PR TITLE
Add random_compat

### LINKED ISSUE #68: Use random_bytes() for key generation!
For PHP 7.0.0 and above, [`random_bytes()`](https://secure.php.net/random_bytes) should be used instead of [`openssl_random_pseudo_bytes()`](https://github.com/paragonie/random_compat/issues/5) (see issue).

We've written a polyfill called [random_compat](https://github.com/paragonie/random_compat) which should polyfill both APIs here. Would it make sense to consolidate all of the CSPRNG effort into that project instead and just add it as a dependency?

### PR DESCRIPTION
**Pull request by paragonie-scott**
We're explicitly NOT using composer for this because we need it to land in the .phar

Fixes #68

### CODE REVIEW

**Comment by paragonie-scott** [comment]
The only question I have is: Do we want it to fallback to `openssl_random_pseudo_bytes()`?

I'd vote no.

**Comment by sstok** [comment]
> We're explicitly NOT using composer for this because we need it to land in the .phar

You can actually use Composer to download the dependency, and then using a Phar stub you make it part of the Phar as normal. This is what is done for many php phar-only projects.

**Comment by paragonie-scott** [comment]
Oh, neat. I'll have to see how that's done then.

**Comment by defuse** [comment]
It could still be nice to include it so that the "clone the repository and then require autoload.php" works. Using composer to fetch random_compat would remove that option, leaving only 1. use composer, or 2. use the .phar.

Actually, I'm totally fine with that. 2 options for how to use this library is simpler than 3!

File: src/random_compat/byte_safe_strings.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+         *
+         * @param string $binary_string
+         * @param int $start
+         * @param int $length (optional)
+         *
+         * @throws TypeError
+         *
+         * @return string
+         */
+        function RandomCompat_substr($binary_string, $start, $length = null)
```
Comment (defuse) [inline src/random_compat/byte_safe_strings.php]: This'll have to be updated with the bugfixes, I think.

Comment (defuse) [inline src/random_compat/byte_safe_strings.php]: Well, actually, it doesn't, because random_compat probably never triggers the bug.

**Comment by defuse** [comment]
Hang on, isn't `mcrypt_create_iv()` available even when you don't actually have mcrypt installed? If that's the case then we should just switch back to using that.

**Comment by defuse** [comment]
Meh, let's just bring in random_compat then we don't have to worry about it.

**Comment by defuse** [comment]
I merged this instead of waiting on @paragonie-scott to add it as a composer dependency because (1) I don't expect it to change much, and (2) I won't have to have/use composer to build the .phar.

**Pull request merged**